### PR TITLE
Removing boost dependency. Now fully cross-platform idomatic c++. Added cmake build with tests.

### PR DESCRIPTION
**Pull request by dribllerrad**
Removing boost dependency. Now fully cross-platform idomatic c++. Added cmake build with tests.

### CODE REVIEW

**Comment by dribllerrad** [comment]
Removing boost dependency. Now fully cross-platform idiomatic c++. Added cmake build with tests. (This does not interfere with your Makefile which still works and builds the tests but it might break that YAML with the `sudo apt install boost` which can probably just be omitted now)

File: src/board.cpp [hunk 1]
```diff
@@ -1,30 +1,33 @@
...
     return this->moves;
 }
 
-Piece* Board::at(string pos) {
-    cout << "checking for duders at " << pos << endl;
+Piece* Board::at(std::string pos) {
+    std::cout << "checking for duders at " << pos << "\n";
     int x = pos[0];
-    int y = boost::lexical_cast<int>(pos[1]);
+    //int y = boost::lexical_cast<int>(pos[1]);
```
Comment (slavaaaaaaaaaa) [inline src/board.cpp]: As git will keep this in its history, feel free to whack these commented out lines.

File: src/board.cpp [hunk 2]
```diff
@@ -1,30 +1,33 @@
...
         return this->matrix[pos[0]-97][pos[1]-1];
 }
 
-bool Board::inBounds(string pos) {
-    cout << "checking for bounds at " << pos << endl;
+bool Board::inBounds(std::string pos) {
+    std::cout << "checking for bounds at " << pos << "\n";
     int x = pos[0];
-    int y = boost::lexical_cast<int>(pos[1]);
+    //int y = boost::lexical_cast<int>(pos[1]);
```
Comment (slavaaaaaaaaaa) [inline src/board.cpp]: same here

File: src/king.cpp [hunk 3]
```diff
@@ -1,19 +1,31 @@
...
-        this->posx + boost::lexical_cast<std::string>(this->posy - 1), // down one
-        static_cast<char>(this->posx - 1) + boost::lexical_cast<std::string>(this->posy), // left one
-        static_cast<char>(this->posx + 1) + boost::lexical_cast<std::string>(this->posy), // right one
-        static_cast<char>(this->posx - 1) + boost::lexical_cast<std::string>(this->posy + 1), // left up
-        static_cast<char>(this->posx + 1) + boost::lexical_cast<std::string>(this->posy + 1), // right up
-        static_cast<char>(this->posx + 1) + boost::lexical_cast<std::string>(this->posy - 1), // right down
-        static_cast<char>(this->posx - 1) + boost::lexical_cast<std::string>(this->posy - 1) // left down
-        //TODO: castling
-    };
+    // std::string moves [] = {
```
Comment (slavaaaaaaaaaa) [inline src/king.cpp]: and all of these

File: src/piece.h [hunk 4]
```diff
@@ -4,38 +4,38 @@
 #include <string>
 #include <vector>
 #include <iostream>
-#include <boost/lexical_cast.hpp>
+//#include <boost/lexical_cast.hpp>
```
Comment (slavaaaaaaaaaa) [inline src/piece.h]: don't need this anymore either

**Review by slavaaaaaaaaaa — COMMENTED** [review]
ready to merge once commented-out code is removed

File: src/piece.h (hunk 4 above)
Comment (dribllerrad) [reply src/piece.h]: Done.

**Pull request merged**